install/oddjob/com.redhat.idm.trust-fetch-domains.in

```diff
@@ -37,7 +37,10 @@ def parse_options():
     options, args = parser.parse_args()
     safe_options = parser.get_safe_opts(options)
 
-    return safe_options, options, args
+    # We only use first argument of the passed args but as D-BUS interface
+    # in oddjobd cannot expose optional, we fill in empty slots from IPA side
+    # and filter them here.
+    return safe_options, options, [args[0]]
```

Comment:
May I propose to merge the argument checking from lines 95 and following into the helper function and have it returned a decoded trusted_domain name? It makes the intent clearer and the code less confusing.

```
if len(args) != 1:
    # LSB status code 2: invalid or excess argument(s)
    raise ScriptError("You must specify trusted domain name", 2)

trusted_domain = ipautil.fsdecode(args[0]).lower()
```

Reply:
Updated. I actually added a check for at least to DNS labels to be present in the domain name because we don't support single label domains for quite some time.